packages/@lwc/ssr-client-utils/src/index.ts

```diff
@@ -56,3 +60,10 @@ class StyleDeduplicator extends HTMLElement {
 export function registerLwcStyleComponent() {
     customElements.define('lwc-style', StyleDeduplicator);
 }
+
+/**
+ * Clears the stylesheet cache. This is useful for testing purposes.
+ */
+export function clearStylesheetCache() {
+    stylesheetCache.clear();
+}
```

Comment:
We have a hacky but established convention for exposing helpers that are only used in Karma tests. [[1]](https://github.com/salesforce/lwc/blob/6f1b7949f10265621c5c4c1f0119a94afa80e456/packages/%40lwc/engine-core/src/framework/check-version-mismatch.ts#L18-L23) [[2]](https://github.com/salesforce/lwc/blob/6f1b7949f10265621c5c4c1f0119a94afa80e456/packages/%40lwc/engine-core/src/framework/fragment-cache.ts#L27-L34) [[3]](https://github.com/salesforce/lwc/blob/6f1b7949f10265621c5c4c1f0119a94afa80e456/packages/%40lwc/engine-core/src/framework/stylesheet.ts#L39-L46) etc.
```suggestion
// Only used in LWC's Karma tests
if (process.env.NODE_ENV === 'test-karma-lwc') {
  (window as any).__lwcClearStylesheetCache = () => stylesheetCache.clear();
}
```